riverdriver/riverpgxv5/internal/dbsqlc/river_job.sql

```diff
@@ -220,8 +223,17 @@ INSERT INTO river_job(
     @queue,
     coalesce(sqlc.narg('scheduled_at')::timestamptz, now()),
     @state,
-    coalesce(@tags::varchar(255)[], '{}')
-) RETURNING *;
+    coalesce(@tags::varchar(255)[], '{}'),
+    @unique_key,
+    @unique_states
+)
+ON CONFLICT (unique_key)
+    WHERE unique_key IS NOT NULL
+      AND unique_states IS NOT NULL
+      AND river_job_state_in_bitmask(unique_states, state)
```

Comment:
I wish there were a better way but AFAICT this is the only way to indicate the conflict given the specific partial index in use here.